torch/nn/modules/module.py

```diff
@@ -1343,20 +1347,25 @@ def register_forward_pre_hook(
                 ``forward_pre`` hooks registered with
                 :func:`register_module_forward_pre_hook` will fire before all
                 hooks registered by this method.
+            with_kwargs (bool): If true the input and output contains the kwargs.
 
         Returns:
             :class:`torch.utils.hooks.RemovableHandle`:
                 a handle that can be used to remove the added hook by calling
                 ``handle.remove()``
         """
         handle = hooks.RemovableHandle(self._forward_pre_hooks)
+        if not with_kwargs:
+            def f(m, inp, kwargs=None):
+                hook(m, inp), kwargs
```

Comment:
missing a return
Also you can use functools.wraps() to make errors nicer

Reply:
👍